### PR TITLE
RFC: Added `iteratoreltype` method to `repeated` 

### PR DESCRIPTION
**Pull request by gasagna**
Also adds some tests.

### CODE REVIEW

**Comment by mschauer** [comment]
There is a fallback https://github.com/JuliaLang/julia/blob/master/base/generator.jl#L47 for this case, question is if we want to rely on the fallback or be explicit where possible.

**Comment by gasagna** [comment]
Well, ok. I did not notice the fallback method. I guess it is up to the someone from the core developers to decide. If we go for being implicit, I can remove that method, and leave the tests.

File: base/iterator.jl [hunk 1]
```diff
@@ -284,19 +284,23 @@ end
 
 done(it::Cycle, state) = state[2]
 
-# Repeat an object infinitely many times
+
+# Repeated - repeat an object `n` or infinitely many times
```
Comment (tkelman) [inline base/iterator.jl]: the type is only for infinite repeats, the lower case method happens to have a convenience method but that will return a Take object wrapping a Repeated

Comment (gasagna) [inline base/iterator.jl]: should it be ditched then?

Comment (tkelman) [inline base/iterator.jl]: the revision to the comment is just not quite accurate, is all

File: base/iterator.jl [hunk 2]
```diff
@@ -284,19 +284,23 @@ end
...
 
-# Repeat an object infinitely many times
+
+# Repeated - repeat an object infinitely many times
 
 immutable Repeated{O}
     x::O
 end
 repeated(x) = Repeated(x)
+repeated(x, n::Integer) = take(repeated(x), n)
```
Comment (tkelman) [inline base/iterator.jl]: I think broadening this to Integer will just give a MethodError unless the same is done for `take`

**Pull request merged**